eservice/lib/libpdo_enclave/contract_request.cpp

```diff
@@ -160,13 +162,20 @@ ContractResponse ContractRequest::process_initialization_request(void)
         pdo::contracts::ContractState new_contract_state;
         std::map<string, string> dependencies;
 
+        sgx_thread_mutex_lock(&worker_->mutex_);
+        worker_->interpreterIsReady();
+        worker_->interpreterBusy();
+
         SAFE_LOG(PDO_LOG_DEBUG, "KV id before interpreter: %s\n",
             ByteArrayToHexEncodedString(contract_state_.state_hash_).c_str());
-        interpreter.set_contract_kv(contract_state_.kv_);
+        interpreter->set_contract_kv(contract_state_.kv_);
 
-        interpreter.create_initial_contract_state(
+        interpreter->create_initial_contract_state(
             contract_id_, creator_id_, code, msg, new_contract_state);
 
+        worker_->interpreterDone();
+        sgx_thread_mutex_unlock(&worker_->mutex_);
```

Comment:
The mutex lock goes away with the above change, just need the call to interpreterDone() (which should internally acquire the lock)